BackendApartmentReservation/BackendApartmentReservation/Apartments/ApartmentRepository.cs

```diff
@@ -11,5 +16,44 @@ public ApartmentRepository(DatabaseContext db)
         {
             _db = db;
         }
+
+        public async Task<DbRoomReservation> CreateRoomReservation(DbEmployee employee, DateTime dateFrom, DateTime dateTo)
+        {
+            var roomReservation = new DbRoomReservation();
+            roomReservation.Employee = employee;
+            roomReservation.DateFrom = dateFrom;
+            roomReservation.DateTo = dateTo;
+
+            var availableRooms = new List<DbApartmentRoom>();
+            foreach (DbApartmentRoom room in _db.ApartmentRooms)
+            {
+                var isAvailable = _db.DbRoomReservations
+                    .Where(e => e.Room.Id == room.Id)
+                    .All(e => e.DateFrom.Date > dateTo.Date || e.DateTo.Date < dateFrom.Date);
```

Comment:
The logic here is wrong. With `All` you are looking if there are any reservations that match criteria.

You should expect that there are **NO** reservations, that are in your provided period

Reply:
One of possible solutions is 
`!Any(r.Room.Id == room.Id && r.DateFrom.Date > ... )`

Reply:
Possible, but not very optimal date filtering
`(to >= r.From && to <= r.To) || (from >= r.From && from <= r.To) || (r.From <= from && r.To >= to)`

Reply:
Or maybe better, this:
https://stackoverflow.com/a/12888920